src/main/java/org/mockito/internal/MockitoCore.java

```diff
@@ -78,16 +80,25 @@ public boolean isTypeMockable(Class<?> typeToMock) {
     }
 
     public <T> T verify(T mock, VerificationMode mode) {
-        if (mock == null) {
-            throw nullPassedToVerify();
-        }
-        if (!isMock(mock)) {
-            throw notAMockPassedToVerify(mock.getClass());
-        }
         MockingProgress mockingProgress = mockingProgress();
         VerificationMode actualMode = mockingProgress.maybeVerifyLazily(mode);
-        mockingProgress.verificationStarted(new MockAwareVerificationMode(mock, actualMode));
-        return mock;
+
+        Set<VerificationListener> listeners = mockingProgress.verificationListeners();
+        try {
+            if (mock == null) {
+                throw nullPassedToVerify();
+            }
+            if (!isMock(mock)) {
+                throw notAMockPassedToVerify(mock.getClass());
+            }
+            mockingProgress.verificationStarted(new MockAwareVerificationMode(mock, actualMode, listeners));
+            return mock;
+        } catch (MockitoException e) {
+            for (VerificationListener listener : listeners) {
+                listener.onVerificationException(mock, actualMode, e);
+            }
```

Comment:
instead of looping here and elsewhere ([1](https://github.com/mockito/mockito/pull/719/files#diff-df53f45103af61fe77db9216591b7ad5R64), [2](https://github.com/mockito/mockito/pull/719/files#diff-60afe350d3acfd74fc0b6a42026e9dc1R34)), I propose to have an internal listener that can do the for loop, this would be easier to test too. Thoughts ?